src/main/kotlin/com/sourcegraph/cody/edit/FixupService.kt

```diff
@@ -60,22 +58,17 @@ class FixupService(val project: Project) : Disposable {
   fun getActiveSession(): FixupSession? = activeSession
 
   fun setActiveSession(session: FixupSession) {
-    if (session == activeSession) return
-    cancelActiveSession()
+    activeSession?.let { if (it.isShowingAcceptLens()) it.accept() else it.cancel() }
+    waitUntilActiveSessionIsFinished()
     activeSession = session
   }
 
-  // Fully cancels/retracts any current session.
-  fun cancelActiveSession() {
-    try {
-      activeSession?.finish()
-    } catch (x: Exception) {
-      logger.warn("Error while disposing session", x)
+  fun waitUntilActiveSessionIsFinished() {
+    while (activeSession != null) {
+      Thread.sleep(100)
```

Comment:
That is kind of bad, and proper fix would be to manage that kind of state purely on the agent side.
But that would require bigger changes, so it is what it is for now.
I think I will add N number of retries there and if it will fail to finish I will use error lenses to communicate an issue.